src/Components/Coins/CoinsButton.js

```diff
@@ -1,6 +1,32 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import CoinsTab from './CoinsTab';
 import './CoinsButton.css';
+import { Button } from '@material-ui/core';
+import { MountedComponentsContext } from '../../Contexts/MountedComponentsContext';
 
 export default function CoinsButton() {
-  return <div id="coinsButton">Coins</div>;
+  const { componentsList, ToggleOn, ToggleOff } = useContext(MountedComponentsContext);
+
+  let isCoinsTabOpen = componentsList.secondaries.CoinsTab;
+  let togglableContent;
+  if (isCoinsTabOpen) {
+    togglableContent = <CoinsTab handleClose={() => ToggleOff(['CoinsTab', 'CoinsPurchaseWindow'])} />;
+  } else togglableContent = null;
+
+  const handleClick = ifMounted => {
+    if (ifMounted) {
+      ToggleOff(['CoinsTab', 'CoinsPurchaseWindow']);
+    } else {
+      ToggleOn(['CoinsTab']);
+    }
+  };
+
+  return (
+    <div id="coinsButtonContainer">
+      <Button onClick={() => handleClick(isCoinsTabOpen)} id="coinsButton">
```

Comment:
`handleClick` can get `isCoinsTabOpen` by itself, it doesn't need to be passed as arg